### PR TITLE
Specify tolerance for individual tests | test(torchlib)

### PR DESCRIPTION
**Pull request by justinchuby**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #792
* __->__ #804

### CODE REVIEW

**Review by thiagocrepaldi — COMMENTED** [review]
In general, it is usually better to have a single (low) tolerance threshold as a constant in the project and try to improve the whole system to have higher accuracy. 

Having successes with low accuracy may hide larger issues that should be tackled sooner than later. Ideally each inaccuracy issue should be tracked by an issue to isolate and fix the root cause.

File: onnxscript/tests/function_libs/torch_lib/ops_test.py [hunk 1]
```diff
@@ -194,6 +181,13 @@ def run_test_output_match(
             f"Type constraints: {onnx_function.op_schema.type_constraints}"
         )
 
+    # Obtain the tolerance for the op if specified
+    if (tolerance := torchlib_op_info.tolerance.get(dtype)) is not None:
+        rtol, atol = tolerance
+    else:
+        rtol = None
```
Comment (BowenBao) [inline onnxscript/tests/function_libs/torch_lib/ops_test.py:188]: What is meaning of None? Please add comment :)

Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test.py:188]: None makes PyTorch to use the default tolerance for the data type. I will add a comment

File: onnxscript/tests/function_libs/torch_lib/ops_test.py [hunk 2]
```diff
@@ -194,6 +181,13 @@ def run_test_output_match(
             f"Type constraints: {onnx_function.op_schema.type_constraints}"
         )
 
+    # Obtain the tolerance for the op if specified
+    if (tolerance := torchlib_op_info.tolerance.get(dtype)) is not None:
+        rtol, atol = tolerance
+    else:
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test.py:187]:
```suggestion
    # Use the PyTorch default if not specified
    # https://pytorch.org/docs/stable/testing.html
    else:
```

File: onnxscript/tests/function_libs/torch_lib/ops_test.py [hunk 3]
```diff
@@ -194,6 +181,13 @@ def run_test_output_match(
             f"Type constraints: {onnx_function.op_schema.type_constraints}"
         )
 
+    # Obtain the tolerance for the op if specified
+    if (tolerance := torchlib_op_info.tolerance.get(dtype)) is not None:
```
Comment (justinchuby) [inline onnxscript/tests/function_libs/torch_lib/ops_test.py:185]: Extract this as a method

**Review by fatcat-z — APPROVED** [review]

**Review by fatcat-z — APPROVED** [review]

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py [hunk 4]
```diff
@@ -77,11 +77,22 @@ class TorchLibOpInfo:
     nondeterministic: bool = False
     # Whether the function is designed for complex inputs
     complex: bool = False
+    # The tolerance for the test {dtype: (rtol, atol)}.
```
Comment (fatcat-z) [inline onnxscript/tests/function_libs/torch_lib/ops_test_data.py:80]: nit, change to:

The tolerance of the inference result difference between PyTorch and ORT. Format: {dtype: (rtol, atol)}.
For example: {torch.float16: (1e-3, 1e-3)}

File: onnxscript/tests/function_libs/torch_lib/ops_test.py (hunk 1 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test.py]: Done

File: onnxscript/tests/function_libs/torch_lib/ops_test_data.py (hunk 4 above)
Comment (justinchuby) [reply onnxscript/tests/function_libs/torch_lib/ops_test_data.py:80]: Done

**Comment by justinchuby** [comment]
Merging this. Happy to follow up if there are more questions

**Pull request merged**